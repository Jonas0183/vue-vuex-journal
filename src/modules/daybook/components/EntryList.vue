<template>        
    <div class="entry-list-container">
        <div class="px-2 pt-2">
            <input 
            type="text"
            class="form-control"
            placeholder="Buscar entradas" 
            v-model="term"
            />
        </div>
        <div class="entry-scroll-area">
            <Entry
            v-for="entry in entriesByTerm"
            :key="entry.id"
            v-model="term"
            :entry="entry"
            />
        </div>
    </div>
</template>
<script>
import { mapGetters } from 'vuex';
import { defineAsyncComponent } from 'vue'

export default {
    components:{
        Entry: defineAsyncComponent(() => import('./Entry.vue'))
    },
    computed: {
        entriesByTerm() {
            return this.getEntriesByTerm( this.term )
        },
        ...mapGetters('journal',['getEntriesByTerm'])
    },
    data() {
        return {
            term:''
        }
    },

}
</script>
<style lang="scss" scoped>

.entry-list-container{
    border-right: 1px solid #2c3e50;
    height: calc(100vh - 56px);

}
.entry-scroll-area{
    height: calc( 100vh - 110px );
    overflow: scroll;

}
</style>