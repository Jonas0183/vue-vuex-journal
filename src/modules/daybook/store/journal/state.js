export default () => ({
    isLoading: true,
    entries:[
        {
            id: '1', //new Date().getTime(),
            date : new Date().toDateString(),
            text: 'Cillum occaecat Lorem deserunt est. Excepteur ut magna voluptate pariatur sit ullamco exercitation sunt excepteur reprehenderit exercitation qui. Aute ullamco est ad aliqua amet duis nisi voluptate minim velit commodo consectetur.',
            picture: null,
        },
        {
            id: '2',//new Date().getTime() + 1000,
            date : new Date().toDateString(),
            text: 'Laboris reprehenderit fugiat labore qui ullamco ex dolore. Reprehenderit consequat culpa irure consequat pariatur ut eiusmod sint quis. Nostrud sit sint reprehenderit dolore voluptate fugiat adipisicing consectetur in.',
            picture: null,
        },
        {
            id: '3',// new Date().getTime() + 2000,
            date : new Date().toDateString(),
            text: 'Cillum proident nulla eiusmod irure ex aliquip laborum. Laborum ut sit magna fugiat deserunt sunt excepteur dolore ad veniam. Nostrud cillum Lorem esse sint nisi. Ad fugiat eiusmod ad qui aute dolore dolor ullamco et. Nostrud elit aliquip labore exercitation cillum fugiat incididunt ea. Officia laboris et minim ad mollit qui cupidatat laborum tempor exercitation. Eiusmod duis tempor Lorem minim culpa velit ut.',
            picture: null,
        },
    ]
})