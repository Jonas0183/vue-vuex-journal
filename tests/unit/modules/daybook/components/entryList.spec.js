import { createStore } from 'vuex'
import { shallowMount } from "@vue/test-utils";

import { journalState } from "../../../mock-data/test-journal-state";

import journal from "@/modules/daybook/store/journal";
import EntryList from "@/modules/daybook/components/EntryList";

const createVuexStore = ( inicialState ) => 
    createStore({
        modules: {
            journal: {
                ...journal,
                state: { ...inicialState }
            }
        }
    })

describe('Pruebas en el componente EntryList', () => {
    
    const store = createVuexStore( journalState )
    const mockRouter = {
        push: jest.fn()
    }

    let wrapper
    beforeEach( () => {

        jest.clearAllMocks()

        wrapper = shallowMount( EntryList, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        } )

    })
    test('Debe llamar al getEntriesByTerm sin termino y mostrar 2 entradas', () => {

        expect( wrapper.findAll('entry-stub').length).toBe(2)
        expect( wrapper.html()).toMatchSnapshot()
    })
    test('Debe llamar al getEntriesByTerm y filtrar las entradas', async () => {

        const input = wrapper.find('input')
        await input.setValue('segunda')

        expect( wrapper.findAll('entry-stub').length).toBe(1)
    })
    test('El boton de nuevo, debe redireccionar a /new', () => {
        wrapper.find('button').trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith( { name: 'entry', params: { id: 'new' } } )
    })

})