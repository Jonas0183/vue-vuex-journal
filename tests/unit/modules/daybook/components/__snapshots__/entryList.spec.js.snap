// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EntryList Debe llamar al getEntriesByTerm sin termino y mostrar 2 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entradas"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle mr-2"></i> Nueva entrada </button></div>
  <div class="entry-scroll-area">
    <entry-stub modelvalue="" entry="[object Object]"></entry-stub>
    <entry-stub modelvalue="" entry="[object Object]"></entry-stub>
  </div>
</div>
`;
